src/tests/ftest/util/dmg_utils.py

```diff
@@ -23,44 +23,183 @@
+        self.debug = FormattedParameter("-d", False)
+        self.json = FormattedParameter("-j", False)
+
+    def _get_action_command(self):
+        """Assign a command object for the specified request and action."""
+        if self.action.value == "format":
+            self.action_command = self.DmgFormatSubCommand()
+        elif self.action.value == "prepare":
+            self.action_command = self.DmgPrepareSubCommand()
+
+    class DmgFormatSubCommand(CommandWithParameters):
+        """Defines an object representing a format sub dmg command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgFormatSubCommand, self).__init__(
+                "/run/dmg/format/*", "format")
+            self.force = FormattedParameter("-f", False)
+
+    class DmgPrepareSubCommand(CommandWithParameters):
+        """Defines a object representing a prepare sub dmg command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgPrepareSubCommand, self).__init__(
+                "/run/dmg/prepare/*", "prepare")
+            self.pci_wl = FormattedParameter("-w {}")
+            self.hugepages = FormattedParameter("-p {}")
+            self.targetuser = FormattedParameter("-u {}")
+            self.nvmeonly = FormattedParameter("-n", False)
+            self.scmonly = FormattedParameter("-s", False)
+            self.force = FormattedParameter("-f", False)
+            self.reset = FormattedParameter("--reset", False)
+
+
+def storage_scan(hosts):
+    """ Execute scan command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run scan on.
+
+    Returns:
+        Avocado CmdResult object that contains exit status, stdout information.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.request.value = "storage"
+    dmg.action.value = "scan"
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+
+    try:
+        result = dmg.run()
+    except CommandFailure as details:
+        print("<dmg> command failed: {}".format(details))
+        return False
+
+    return result
+
+
+def storage_format(hosts):
+    """ Execute format command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run format on.
+
+    Returns:
+        Avocado CmdResult object that contains exit status, stdout information.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+    dmg.request.value = "storage"
+    dmg.action.value = "format"
+    dmg._get_action_command()
+    dmg.action_command.force.value = True
+
+    try:
+        result = dmg.run(sudo=True)
+    except CommandFailure as details:
+        print("<dmg> command failed: {}".format(details))
+        return False
+
+    return result
```

Comment:
Keeping in line with the 'helper' nature of this function, shouldn't it return True here (assuming a rc=0 is sufficient to indicate success), or if needed parse the stdout/stderr from the CmdResult to indicate success or failure?

Reply:
If we rely on the return code we wouldn't be able to tell why it failed or which server failed.